### PR TITLE
Implement tracking validation

### LINKED ISSUE #8: Public website: Validation of tracking number
- Verify the tracking number is valid
- If it's valid display the information about the shipment.

### PR DESCRIPTION
**Pull request by cdsandoval**
Close #8 

Add validation for tracking id number.

### CODE REVIEW

File: app/controllers/shipment_controller.rb [hunk 1]
```diff
@@ -1,6 +1,22 @@
 class ShipmentController < ApplicationController
   
   def show
+    @shipment = Shipment.find_by_tracking_id(params[:tracking_id])
+
+    if @shipment == nil
```
Comment (Linzeur) [inline app/controllers/shipment_controller.rb]: I recommend you to use "unless @shipment" instead of that

Comment (cdsandoval) [reply app/controllers/shipment_controller.rb]: Good idea, thanks for your advice. I'll refactor these lines of code.

File: app/views/shipment/show.html.erb [hunk 2]
```diff
@@ -1 +1,13 @@
...
+<li><%= @shipment.origin_adress %></li>
+<li><%= @shipment.destination_address %></li>
+<li><%= @shipment.weight %></li>
+<li><%= @shipment.reception_date %></li>
+<li><%= @shipment.estimated_delivery_date %></li>
+<li><%= @shipment.delivered_date %></li>
+<li><%= @shipment.freight_value %></li>
+<li><%= @shipment.user.username %></li>
+<li><%= @shipment.sender.store_name %></li>
+
```
Comment (Linzeur) [inline app/views/shipment/show.html.erb]: You should remember that not all fields will be shown when you do a query, it depends on the email's user logged

Comment (cdsandoval) [reply app/views/shipment/show.html.erb]: In these issues all I need to do is validate if tracking_id exist or not?. Those validations you are mention if will add later.

File: app/controllers/shipment_controller.rb [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+      flash[:notice] = "The shipment doesn't exist"
+      redirect_to "/"
```
Comment (yummta) [inline app/controllers/shipment_controller.rb]: Avoid use hardcode in the redirect "/". Instead, use 'root_path'.

File: app/views/shipment/show.html.erb (hunk 2 above)
Comment (Linzeur) [reply app/views/shipment/show.html.erb]: Excellent

File: app/controllers/shipment_controller.rb (hunk 3 above)
Comment (cdsandoval) [reply app/controllers/shipment_controller.rb]: Woah, you're right. Thanks you. I already changed it.

**Pull request merged**